src/linux/mod.rs

```diff
@@ -85,3 +92,99 @@ impl Sandbox for LinuxSandbox {
         }
     }
 }
+
+/// Restrict networking using seccomp and namespaces.
+fn restrict_networking() -> Result<()> {
+    // Create network namespace.
+    let result = create_user_namespace(true).and_then(|_| unshare(Namespaces::NETWORK));
+
+    // Apply seccomp network filter.
+    match NetworkFilter::apply() {
+        _ if result.is_ok() => Ok(()),
+        result => result,
+    }
+}
+
+/// Create a new user namespace.
+///
+/// If the `become_root` flag is set, then the current user will be mapped to
+/// UID 0 inside the namespace. Otherwise the current user will be mapped to its
+/// UID of the parent namespace.
+fn create_user_namespace(become_root: bool) -> Result<()> {
+    // Get the current UID.
+    let uid = unsafe { libc::getuid() };
+
+    // Create the namespace.
+    unshare(Namespaces::USER)?;
+
+    // Map the UID and GID.
+    let map = if become_root { format!("0 {uid} 1\n") } else { format!("{uid} {uid} 1\n") };
+    fs::write("/proc/self/uid_map", map.as_bytes())?;
+    fs::write("/proc/self/setgroups", b"deny")?;
+    fs::write("/proc/self/gid_map", map.as_bytes())?;
```

Comment:
Writing the same map to `gid_map` implies that the current UID and GID are the same... That might be true in the typical case, but it seems like a very risky assumption...

Reply:
Very good point.